### ISSUE TITLE
How to reproduce the results of summarization(Rouge1-39.84) with dynamic convolution in the Pay Less attention paper?

### ISSUE THREAD

**Issue by hong8c** [pytorch/fairseq#594]
Hi
I asked the same question a few days ago. But I cannot reproduce the results(rouge1-39.84) on the summarization task in "Pay less attention with Lightweight and Dynamic convolutions" paper.

I run the code with these parameters below and got the results with rouge1-34.18. Furthermore, the model with 1024(word embedding dim) has about 150M training parameters, not 87M parameters in the paper reported.
Could you please tell me **_what is the training and inference arguments_** used in the paper so that i can get the same results with the model "DynamicConv 87M params"?

Thank you~!

python -u $code_dir/train.py \
${dataset_dir} \
--save-dir ${model_dir} \
--tensorboard-logdir $model_dir \
--task translation \
--seed 1 \
--num-workers 3 \
--max-tokens 3584 \
--arch lightconv \
--criterion label_smoothed_cross_entropy \
--update-freq 16 \
--optimizer adam \
--lr '1e-7' \
--clip-norm 0.0 \
--momentum 0.99 \
--weight-decay 0.001 \
--lr-scheduler cosine \
--lr-shrink 1 \
--min-lr '1e-09' \
--max-lr 0.001 \
--min-loss-scale 0.0001 \
--label-smoothing 0.1 \
--save-interval-updates 150 \
--no-epoch-checkpoints \
--keep-interval-updates 5 \
--lr-period-updates 20000 \
--warmup-updates 10000 \
--warmup-init-lr '1e-07' \
--adam-betas '(0.9, 0.98)' \
--dropout 0.3 \
--attention-dropout 0.1 \
--relu-dropout 0.0 \
--encoder-embed-dim 512 \
--encoder-ffn-embed-dim 2048 \
--encoder-layers 7 \
--encoder-attention-heads 8 \
--decoder-embed-dim 512 \
--decoder-ffn-embed-dim 2048 \
--decoder-layers 6 \
--decoder-attention-heads 8 \
--share-all-embeddings \
--input-dropout 0.1 \
--encoder-kernel-size-list [3,7,15,31,31,31,31] \
--decoder-kernel-size-list [3,7,15,31,31,31] \
--encoder-glu 1 \
--decoder-glu 1 \
--encoder-conv-type dynamic \
--decoder-conv-type dynamic \
--weight-softmax True \
--weight-dropout 0.1 \
--no-progress-bar \
--log-interval 50

**Comment by huihuifan** [comment]
Did you train with BPE? We learn 40k bpe units to preprocess the dataset. 

For generation, we set lenpen to 0.9, max len to 200, min len to 35, and did not allow n = 3 grams to repeat.

**Comment by MichaelZhouwang** [comment]
> Hi
> I asked the same question a few days ago. But I cannot reproduce the results(rouge1-39.84) on the summarization task in "Pay less attention with Lightweight and Dynamic convolutions" paper.
> 
> I run the code with these parameters below and got the results with rouge1-34.18. Furthermore, the model with 1024(word embedding dim) has about 150M training parameters, not 87M parameters in the paper reported.
> Could you please tell me **_what is the training and inference arguments_** used in the paper so that i can get the same results with the model "DynamicConv 87M params"?
> 
> Thank you~!
> 
> python -u $code_dir/train.py 
> ${dataset_dir} 
> --save-dir ${model_dir} 
> --tensorboard-logdir $model_dir 
> --task translation 
> --seed 1 
> --num-workers 3 
> --max-tokens 3584 
> --arch lightconv 
> --criterion label_smoothed_cross_entropy 
> --update-freq 16 
> --optimizer adam 
> --lr '1e-7' 
> --clip-norm 0.0 
> --momentum 0.99 
> --weight-decay 0.001 
> --lr-scheduler cosine 
> --lr-shrink 1 
> --min-lr '1e-09' 
> --max-lr 0.001 
> --min-loss-scale 0.0001 
> --label-smoothing 0.1 
> --save-interval-updates 150 
> --no-epoch-checkpoints 
> --keep-interval-updates 5 
> --lr-period-updates 20000 
> --warmup-updates 10000 
> --warmup-init-lr '1e-07' 
> --adam-betas '(0.9, 0.98)' 
> --dropout 0.3 
> --attention-dropout 0.1 
> --relu-dropout 0.0 
> --encoder-embed-dim 512 
> --encoder-ffn-embed-dim 2048 
> --encoder-layers 7 
> --encoder-attention-heads 8 
> --decoder-embed-dim 512 
> --decoder-ffn-embed-dim 2048 
> --decoder-layers 6 
> --decoder-attention-heads 8 
> --share-all-embeddings 
> --input-dropout 0.1 
> --encoder-kernel-size-list [3,7,15,31,31,31,31] 
> --decoder-kernel-size-list [3,7,15,31,31,31] 
> --encoder-glu 1 
> --decoder-glu 1 
> --encoder-conv-type dynamic 
> --decoder-conv-type dynamic 
> --weight-softmax True 
> --weight-dropout 0.1 
> --no-progress-bar 
> --log-interval 50

Have you succeeded in reproducing the results by using this hyperparameter with BPE? Thanks

**Comment by lioutasb** [comment]
Is there a script to preprocess the CNN/DM dataset with Fairseq? I want to reproduce the result reported in the paper.
